src/impl_/wrap.rs

```diff
@@ -40,6 +44,9 @@ where
     T: IntoPy<PyObject>,
 {
     type Error = E;
+    fn wrap_no_gil(self) -> Result<T, Self::Error> {
+        self
+    }
```

Comment:
Hmm, I think this adds some complexity to the `OkWrap` trait which we didn't previously have, I'll push an alternative, let's see what we think of the alternative.

Reply:
I've rebased on main.